acceptance-tests/src/test-support/java/tech/pegasys/artemis/test/acceptance/dsl/ArtemisNode.java

```diff
@@ -278,59 +280,41 @@ public void captureDebugArtifacts(final File artifactDir) {
-    private static final String BEACONRESTAPI_SECTION = "beaconrestapi";
-    private static final String DEPOSIT_SECTION = "deposit";
-    private static final String INTEROP_SECTION = "interop";
-    private static final String NODE_SECTION = "node";
-    private static final String VALIDATOR_SECTION = "validator";
     private static final String VALIDATORS_FILE_PATH = "/validators.yml";
-    private static final String OUTPUT_SECTION = "output";
-    private Map<String, Map<String, Object>> options = new HashMap<>();
+    private static final String P2P_PRIVATE_KEY_FILE_PATH = "/p2p-private-key.key";
     private static final int DEFAULT_VALIDATOR_COUNT = 64;
 
+    private ArtemisConfigurationBuilder artemisConfigurationBuilder;
+
     private Optional<String> validatorKeys = Optional.empty();
     private Optional<GenesisStateConfig> genesisStateConfig = Optional.empty();
+    private boolean p2pEnabled = false;
 
     public Config() {
-      final Map<String, Object> node = getSection(NODE_SECTION);
-      setNetworkMode("mock");
-      node.put("networkInterface", NetworkUtility.INADDR_ANY);
-      node.put("port", P2P_PORT);
-      node.put("discovery", "static");
-      node.put("constants", "minimal");
-
-      final Map<String, Object> interop = getSection(INTEROP_SECTION);
-      interop.put("genesisTime", 0);
-      interop.put("ownedValidatorStartIndex", 0);
-      interop.put("ownedValidatorCount", DEFAULT_VALIDATOR_COUNT);
-
-      final Map<String, Object> deposit = getSection(DEPOSIT_SECTION);
-      setDepositMode("test");
-      deposit.put("numValidators", DEFAULT_VALIDATOR_COUNT);
-
-      final Map<String, Object> beaconRestApi = getSection(BEACONRESTAPI_SECTION);
-      beaconRestApi.put("portNumber", REST_API_PORT);
-      beaconRestApi.put("enableSwagger", false);
-
-      final Map<String, Object> output = getSection(OUTPUT_SECTION);
-      output.put("transitionRecordDir", ARTIFACTS_PATH + "transitions/");
-
-      final Map<String, Object> database = getSection(DATABASE_SECTION);
-      database.put("dataDir", ARTIFACTS_PATH + "data/");
+      artemisConfigurationBuilder =
+          ArtemisConfiguration.builder()
+              .setNetwork("minimal")
+              .setP2pEnabled(false)
+              .setP2pDiscoveryEnabled(false)
+              .setP2pPort(P2P_PORT)
+              .setP2pAdvertisedPort(P2P_PORT)
+              .setP2pInterface(NetworkUtility.INADDR_ANY)
+              .setxInteropGenesisTime(0)
+              .setxInteropOwnedValidatorStartIndex(0)
+              .setxInteropOwnedValidatorCount(DEFAULT_VALIDATOR_COUNT)
+              .setxInteropNumberOfValidators(DEFAULT_VALIDATOR_COUNT)
+              .setxInteropEnabled(true)
+              .setRestApiEnabled(true)
+              .setRestApiPort(REST_API_PORT)
+              .setRestApiDocsEnabled(false)
+              .setxTransitionRecordDirectory(ARTIFACTS_PATH + "transitions/")
+              .setDataPath(ARTIFACTS_PATH + "data/");
```

Comment:
We had deliberately kept the code for config separate between production and acceptance tests because we want the ATs to fail if we change a config option.  Breaking backwards compatibility should be a deliberate choice but is easy to do without noticing if the code is shared and the ATs automatically pick up the new config.

Reply:
fixed